src/braket/experimental/algorithms/adaptive_shot_allocation/utils.py

```diff
@@ -0,0 +1,261 @@
+
+from typing import List, Union
```

Comment:
Some functions in this file are helper functions for notebook. These functions should be moved to the notebook folder alongside the notebook. The functions here should be used for the algorithm (i.e., the one used in adaptive_allocator.py)